src/execution_plan/ops/op_value_hash_join.c

```diff
@@ -0,0 +1,289 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_value_hash_join.h"
+#include "../../value.h"
+#include "../../util/arr.h"
+#include "../../util/qsort.h"
+#include <assert.h>
+
+/* Determins order between two records by inspecting
+ * element stored at postion idx. */
+static bool _record_islt(Record l, Record r, uint idx) {
+    SIValue lv = Record_GetScalar(l, idx);
+    SIValue rv = Record_GetScalar(r, idx);
+    return SIValue_Order(lv, rv) < 0;
```

Comment:
Why is this `SIValue_Order` rather than `SIValue_Compare`?

Reply:
because there's no guarantee a.v is of the same type as b.v
in which case you want to compare using type order.